### ISSUE TITLE
multiple identifier search does not work

### ISSUE THREAD

**Issue by mingdatacom** [jamesagnew/hapi-fhir#1223]
Hello All,

In HAPI FHIR 3.6 this worked
http://hapi.fhir.org/baseDstu3/Organization?identifier=FDS-BusinessId%7C965214&identifier=2.16.840.1.113883.3.79%7C1000005
returned Bundle with 1 result

In HAPI FHIR 3.7 Bundle with 0 result
Was there a change in the spec or is there a bug?

Thanks,

**Comment by jamesagnew** [comment]
Having multiple `identifier` parameters is an AND search, so that will only match organizations that have both of those identifiers. Are you sure you aren't looking for an OR search? e.g.

http://hapi.fhir.org/baseDstu3/Organization?identifier=FDS-BusinessId%7C965214,2.16.840.1.113883.3.79%7C1000005

**Comment by dmitchellmim** [comment]
I think it's an issue with all AND searches. There are countless patients on the test server named John Smith, but [Patient?name=john&name=smith](http://hapi.fhir.org/baseDstu3/Patient?name=john&name=smith) returns 0 results. [Patient?given=john&name=smith](http://hapi.fhir.org/baseDstu3/Patient?given=john&name=smith) and [Patient?name=john&family=smith](http://hapi.fhir.org/baseDstu3/Patient?name=john&family=smith) return paged results; it only fails when using `name` for both.

**Comment by mingdatacom** [comment]
Yes we are looking for an AND search i.e an organization having both these identifiers, this is the result using one identifier (http://hapi.fhir.org/baseDstu3/Organization?identifier=FDS-BusinessId%7C965214) :
```
"entry": [
    {
      "fullUrl": "http://hapi.fhir.org/baseDstu3/Organization/941836",
      "resource": {
        "resourceType": "Organization",
        "id": "941836",
        "meta": {
          "versionId": "22",
          "lastUpdated": "2019-03-01T19:54:12.052+00:00"
        },
        "identifier": [
          {
            "system": "FDS-BusinessId",
            "value": "965214"
          },
          {
            "system": "2.16.840.1.113883.3.79",
            "value": "1000005"
          },
          {
            "system": "http://hl7.org/fhir/sid/us-npi"
          }
        ],
        "active": true,
        "name": "Engage Demo Pharmacy",
        "telecom": [
          {
            "system": "phone",
            "value": "410-867-5309",
            "use": "work",
            "rank": 1
          }
        ],
        "address": [
          {
            "text": "  , , "
          }
        ]
      },
      "search": {
        "mode": "match"
      }
    }
  ]
```

**Comment by mingdatacom** [comment]
Steps to verify it works on 3.6:
1. Download https://github.com/jamesagnew/hapi-fhir/releases/download/v3.6.0/hapi-fhir-3.6.0-cli.zip
2. Extract and execute `hapi-fhir-cli run-server -v dstu3`
3. Create organization [create-organization-2.txt](https://github.com/jamesagnew/hapi-fhir/files/2928816/create-organization-2.txt)
  ```
  curl -vX POST http://localhost:8080/baseDstu3/Organization --header "Content-Type: application/json" -d @create-organization-2.txt
  ```
4. Search organization
  ```
  curl -s "http://localhost:8080/baseDstu3/Organization?identifier=FDS-BusinessId%7C965214&identifier=2.16.840.1.113883.3.79%7C1000005"  
  {
  "resourceType": "Bundle",
  "id": "3788da5b-c68a-410d-af62-b5064e3091fd",
  "meta": {
    "lastUpdated": "2019-03-05T08:51:53.517+08:00"
  },
  "type": "searchset",
  "total": 1,
  "link": [
    {
      "relation": "self",
      "url": "http://localhost:8080/baseDstu3/Organization?identifier=FDS-BusinessId%7C965214&identifier=2.16.840.1.113883.3.79%7C1000005"
    }
  ],
  "entry": [
    {
      "fullUrl": "http://localhost:8080/baseDstu3/Organization/9952",
      "resource": {
        "resourceType": "Organization",
        "id": "9952",
        "meta": {
          "versionId": "1",
          "lastUpdated": "2019-03-05T08:51:05.376+08:00"
        },
        "identifier": [
          {
            "system": "FDS-BusinessId",
            "value": "965214"
          },
          {
            "system": "2.16.840.1.113883.3.79",
            "value": "1000005"
          },
          {
            "system": "http://hl7.org/fhir/sid/us-npi"
          }
        ],
        "active": true,
        "name": "Engage Demo Pharmacy",
        "telecom": [
          {
            "system": "phone",
            "value": "410-867-5309",
            "use": "work",
            "rank": 1
          }
        ],
        "address": [
          {
            "text": "  , , "
          }
        ]
      },
      "search": {
        "mode": "match"
      }
    }
  ]
}
  ```

**Comment by jamesagnew** [comment]
Confirmed and fixed in #1231
